Source/WebCore/bindings/js/WindowProxy.cpp

```diff
@@ -58,20 +60,31 @@ WindowProxy::WindowProxy(AbstractFrame& frame)
 {
 }
 
+WindowProxy::WindowProxy(RefPtr<RemoteDOMWindow> window)
+    : m_remoteWindow(window)
+    , m_jsWindowProxies(makeUniqueRef<ProxyMap>())
+{
+}
+
 WindowProxy::~WindowProxy()
 {
     ASSERT(!m_frame);
     ASSERT(m_jsWindowProxies->isEmpty());
 }
 
+Ref<WindowProxy> WindowProxy::create(RefPtr<RemoteDOMWindow> window)
+{
+    return adoptRef(*new WindowProxy(window));
+}
+
 AbstractFrame* WindowProxy::frame() const
 {
     return m_frame.get();
 }
 
 void WindowProxy::detachFromFrame()
 {
-    ASSERT(m_frame);
+    ASSERT(m_frame || m_remoteWindow);
```

Comment:
I don't really get this. A remote window should have a remote frame and thus shouldn't require special handling?